extensions/wikia/ArticleComments/classes/ArticleCommentsAjax.class.php

```diff
@@ -198,36 +200,41 @@ static public function axPost() {
 
 		$response = ArticleComment::doPost( self::getConvertedContent( $wgRequest->getVal( 'wpArticleComment' ) ), $wgUser, $title, $parentId );
 
-		if ( $response !== false ) {
-			if (
-				$title->getNamespace() == NS_USER_TALK &&
-				$response[0] == EditPage::AS_SUCCESS_NEW_ARTICLE &&
-				$title->getText() != $wgUser->getName()
-			) {
-				$user = User::newFromName( $title->getText() );
-
-				if ( $user ) {
-					$user->setNewtalk( true );
-				}
+		if ( $response === false ) {
+			return $result;
+		}
+
+		if (
+			$title->getNamespace() == NS_USER_TALK &&
+			$response[0] == EditPage::AS_SUCCESS_NEW_ARTICLE &&
+			$title->getText() != $wgUser->getName()
+		) {
+			$user = User::newFromName( $title->getText() );
+
+			if ( $user ) {
+				$user->setNewtalk( true );
 			}
+		}
 
-			$listing = ArticleCommentList::newFromTitle( $title );
-			$countAll = $wgLang->formatNum( $listing->getCountAllNested() );
-			$commentsHTML = $response[2]['text'];
+		$listing = ArticleCommentList::newFromTitle( $title );
 
-			$result = array( 'text' => $commentsHTML, 'counter' => $countAll );
+		/** @var Language $wgLang */
+		$countAll = $wgLang->formatNum( $listing->getCountAllNested() );
+		$commentsHTML = $response[2]['text'];
 
-			if ( F::app()->checkskin( 'wikiamobile' ) ) {
-				$result['counterMessage'] = wfMessage( 'wikiamobile-article-comments-counter' )
-					->params( $countAll )
-					->text();
-			}
+		$result = [
+			'text' => $commentsHTML,
+			'counter' => $countAll
+		];
 
-			if ( $parentId ) {
-				$result['parentId'] = $parentId;
-			}
+		if ( F::app()->checkskin( 'wikiamobile' ) ) {
+			$result['counterMessage'] = wfMessage( 'wikiamobile-article-comments-counter' )
+				->params( $countAll )
+				->text();
```

Comment:
I didn't update this, just changed indentation.  It doesn't look like this has any HTML in it so seems safe to escape.